xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -533,6 +550,42 @@ public void initialize() throws InitializationException
         getTemplateRootPath();
     }
 
+    private void checkRequirements(Template template) throws Exception
+    {
+        ComponentManager componentManager = this.componentManagerProvider.get();
+
+        List<Throwable> causes = null;
+        TemplateContent templateContent = template.getContent();
+        for (String propertyName : templateContent.getPropertyNames()) {
+            if (propertyName.startsWith(PROPERTY_REQUIRE_PREFIX)) {
+                String requirementKey = propertyName.substring(PROPERTY_REQUIRE_PREFIX.length());
+
+                if (componentManager.hasComponent(TemplateRequirement.class, requirementKey)) {
+                    try {
+                        TemplateRequirement requirement =
+                            componentManager.getInstance(TemplateRequirement.class, requirementKey);
+
+                        requirement.checkRequirement(requirementKey,
+                            templateContent.getProperty(propertyName, (String) null), template);
+                    } catch (Exception e) {
+                        if (causes == null) {
+                            causes = new ArrayList<>();
+                        }
+
+                        causes.add(e);
+                    }
+                } else {
+                    this.logger.warn("Now template requirement handler could be found for key [{}] in template [{}]",
```

Comment:
Typo, should be "No". I'm also wondering if we shouldn't do more than logging a warning if the requirement couldn't be checked (like failing the requirements check).

Reply:
> I'm also wondering if we shouldn't do more than logging a warning if the requirement couldn't be checked

I hesitated for that, but this warning is only when no handled exist at all, if the handler exist but cannot be used for any reason it will fail the requirement check. This allows things like putting a requirement in an extension for a handler more recent that the minimum version supported by the extension.